src/pages/dashboard/Explore.tsx

```diff
@@ -96,48 +97,56 @@ const Explore = () => {
+    searchTerms.set('page', '' + value);
+    history.push('?' + searchTerms.toString());
+  };
+  const createQueryFromUrl = (location: any) => {
+    const searchTerms = new URLSearchParams(location.search);
+    const institutions = searchTerms.getAll(SearchParameters.institution);
+    const resourceTypes = searchTerms.getAll(SearchParameters.resourceType);
+    const tags = searchTerms.getAll(SearchParameters.tag).map((tag) => tag.toLowerCase());
+    const pageTerm = searchTerms.get(SearchParameters.page);
+    const showInaccessibleParameter = searchTerms.get(SearchParameters.showInaccessible);
+    const offset = pageTerm && Number(pageTerm) !== firstPage ? (Number(pageTerm) - 1) * NumberOfResultsPrPage : 0;
+    const showInaccessible = showInaccessibleParameter ? showInaccessibleParameter.toLowerCase() === 'true' : false;
+    const licenses = searchTerms.getAll(SearchParameters.license);
+    return {
+      ...emptyQueryObject,
+      query: searchTerms.get(SearchParameters.query) ?? '',
+      offset: offset,
+      limit: NumberOfResultsPrPage,
+      resourceTypes: resourceTypes,
+      institutions: institutions,
+      licenses: licenses,
+      tags: tags,
+      showInaccessible: showInaccessible,
+    };
   };
 
   useEffect(() => {
-    const createQueryFromUrl = () => {
-      const searchTerms = new URLSearchParams(location.search);
-      const institutions = searchTerms.getAll(SearchParameters.institution);
-      const resourceTypes = searchTerms.getAll(SearchParameters.resourceType);
-      const tags = searchTerms.getAll(SearchParameters.tag).map((tag) => tag.toLowerCase());
-      const pageTerm = searchTerms.get(SearchParameters.page);
-      const showInaccessibleParameter = searchTerms.get(SearchParameters.showInaccessible);
-      const showInaccessible = showInaccessibleParameter ? showInaccessibleParameter.toLowerCase() === 'true' : false;
-      const licenses = searchTerms.getAll(SearchParameters.license);
-      const offset = pageTerm && Number(pageTerm) !== firstPage ? (Number(pageTerm) - 1) * NumberOfResultsPrPage : 0;
-      return {
-        ...emptyQueryObject,
-        query: searchTerms.get(SearchParameters.query) ?? '',
-        offset: offset,
-        limit: NumberOfResultsPrPage,
-        resourceTypes: resourceTypes,
-        institutions: institutions,
-        licenses: licenses,
-        tags: tags,
-        queryFromURL: true,
-        allowSearch: true,
-        showInaccessible: showInaccessible,
-      };
-    };
-    if (!queryObject.queryFromURL && !queryObject.allowSearch) {
-      setQueryObject(createQueryFromUrl());
+    const newQueryObject = createQueryFromUrl(location);
+    if (!queryFromURL && !allowSearch) {
+      setQueryObject(newQueryObject);
+      setQueryFromURL(true);
+      setAllowSearch(true);
     }
-  }, [location, queryObject.allowSearch, queryObject.queryFromURL]);
+  }, [location, allowSearch, queryFromURL, queryObject]);
```

Comment:
hvis endring av query params fører til unmount og ny mount trenger ikke alt dette gjennom en useEffect, siden de alltid bare vil ha en verdi. Når de da endres begynner man på nytt igjen.
Vet ikke om det er tilfellet, eller om det evt kan trigges selv, men i så fall slipper man å oppdatere verdier flere ganger i useEffect som bare trigger flere nye rerenders

Reply:
Endret denne ganske drastisk, nå leser den kun inn url params, når den første gang, mounter og når man trykker på bakover / fremover knappen. Ellers så kjøres søk fra queryObject direkte. Komponentene som setter queryObject er selv ansvarlig for å legge søke parametere i URL.